Add useFeatureTable and configurable portal url

mpriour opened the pull request:
Addresses the following:
* Update the deps and pre-commit hook so that it builds in yarn v1 and npm v6 and npm v7.
* Add the ability to optionally configure what portal you load items and layers from.
* Add a `useFeatureTable` hook that renders a feature table widget with a passed layer.
* Add a FeatureTable example

mpriour committed:
Make portal environment configurable when loading items

mpriour committed:
Make project compile, update outdated deps, yarn1 AND npm7 compatibility

mpriour committed:
Add the ability to load a Feature Table with options

mpriour committed:
Add examples for a simple stand-alone feature table

mpriour committed:
Export feature table for use in built library

mpriour committed:
Return feature table from both paths of loadTable

mpriour commented on package.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "0.2.0",
+  "version": "0.3.0",
```
Bump version to reflect addition of functionality.

mpriour commented on package.json:
```diff
@@ -24,11 +24,6 @@
     "esri-loader": "^2.13.0",
     "react": ">=16"
   },
-  "husky": {
```
Pre-commit hook is now a file in the `.husky` directory.

mpriour commented on tsconfig.json:
```diff
@@ -7,7 +7,7 @@
     "declaration": true,
     "sourceMap": true,
-    "rootDir": "./",
+    "rootDir": "./src",
```
Strongly recommended by `tsdx` commands.

mpriour commented on .husky/pre-commit:
```diff
@@ -0,0 +1 @@
+npx --prefer-offline tsdx lint
```
npm 7 fundamentally changed the way that scripts are run from other libraries and breaks the way that husky previously worked with the package.json configuration. This change is documented in the husky docs and putting the command here is what is now required. This still works on npm v6 and yarn v1.
The `--prefer-offline` option prevents tsdx from being downloaded just to run the script but would allow it if the tsdx bin couldn't be found.

tomwayson reviewed (changes requested):
Thanks @mpriour, this is great. I appreciate the tooling update as well.

I'm all for it, but I'd like to get netlify working again. Looks like it defaults to node 10, which upset some of the new deps. Can you please add another commit w/ [a `.nvmrc` file](https://docs.netlify.com/configure-builds/manage-dependencies/#node-js-and-javascript)? I'd say set it 14 for now.

Also, I appreciate that you added tests and an example page, but it would be nice to add info about the new hook to the README as well.

mpriour commented:
@tomwayson - thanks for the review. I'll take care of those changes today.

mpriour committed:
Add .nvmrc file to tell netlify what version of node to use

mpriour committed:
Attempt to get example working by using a version of node that is compatible with @babel/compat-data 7.8.6

mpriour committed:
Include new portalUrl option and useFeatureTable in readme

mpriour committed:
useFeatureTable - make the options object and args match the docs

mpriour commented:
I've: 
* added the .nvmrc file, it had to be pegged to version 13.12 of node
* added info to the readme
* updated the feature table code to match the docs and have only 2 parameters not 3. Previously the third one was the `portalUrl`. That has been rolled into the `options` object like in the MapView and SceneView hooks and documented.

tomwayson reviewed (approved):
thanks @mpriour!
